### ISSUE TITLE
fitToCoordinates don't respect the mapPadding provided (android)

### ISSUE THREAD

**Issue by diegorodriguesvieira** [react-native-maps/react-native-maps#3308]
# Bug
When I add the `mapPadding` prop to the map, the` fitToCoordinates` method does not respect the value provided.

It works on iOS, but it doesn't work on Android.

## Environment info

React native info output:

```bash
System:
    OS: macOS 10.15.3
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 916.18 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 10.16.2 - /usr/local/bin/node
    Yarn: 1.19.1 - /usr/local/bin/yarn
    npm: 6.13.7 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.6010548
    Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
  npmPackages:
    react: ~16.9.0 => 16.9.0 
    react-native: https://github.com/expo/react-native/archive/sdk-36.0.0.tar.gz => 0.61.4 
  npmGlobalPackages:
    react-native-cli: 2.0.1
```

Library version: x.x.x

## Steps To Reproduce

1. add mapPadding to `<MapView />`
2. call method `fitToCoordinates`
3. the method `fitToCoordinates` will not respect the padding provided (android)
...

Describe what you expected to happen:

1. add mapPadding to `<MapView />`
2. call method `fitToCoordinates`
3. the method `fitToCoordinates` should respect the padding provided

## Reproducible sample code
https://snack.expo.io/@devlinkle/maps

Android:

![Screenshot_2020-02-26-16-29-23-274_host exp exponent](https://user-images.githubusercontent.com/7308074/75381032-ca53c080-58b6-11ea-8443-66343dc47997.jpg)

iOS:
<img width="566" alt="Screen Shot 2020-02-26 at 16 25 11" src="https://user-images.githubusercontent.com/7308074/75381048-d2136500-58b6-11ea-94ba-d9757e162aaa.png">

**Comment by diegorodriguesvieira** [comment]
First of all I would like to say that I don't understand about the package's core code. But I did my best to try to find out why `EdgePadding` is miscalculated on android.

The first thing I discovered is that all javaScript functions that use the `EdgePadding` parameter have the same problem on android. This is because the pixel density of the device is not being considered. Another thing, `mapPadding` seems to be fixed at `50`.

The `mapPadding` prop passed to `<MapView />` works perfectly on android. But why? Because it considers pixel density!
https://github.com/react-native-community/react-native-maps/blob/master/lib/android/src/main/java/com/airbnb/android/react/maps/AirMapManager.java#L129-L156

My problem was with the `fitToCoordinates` function, I made the changes to consider the density:

```Java
  public void fitToCoordinates(ReadableArray coordinatesArray, ReadableMap edgePadding, boolean animated) {
    if (map == null)
      return;

    LatLngBounds.Builder builder = new LatLngBounds.Builder();

    for (int i = 0; i < coordinatesArray.size(); i++) {
      ReadableMap latLng = coordinatesArray.getMap(i);
      Double lat = latLng.getDouble("latitude");
      Double lng = latLng.getDouble("longitude");
      builder.include(new LatLng(lat, lng));
    }

    LatLngBounds bounds = builder.build();
    CameraUpdate cu = CameraUpdateFactory.newLatLngBounds(bounds, baseMapPadding);

    final double density = (double) getContext().getResources().getDisplayMetrics().density;

    if (edgePadding != null) {
      map.setPadding((int) (edgePadding.getInt("left") * density), (int) (edgePadding.getInt("top") * density),
          (int) (edgePadding.getInt("right") * density), (int) (edgePadding.getInt("bottom") * density));
    }

    if (animated) {
      map.animateCamera(cu);
    } else {
      map.moveCamera(cu);
    }
    map.setPadding(0, 0, 0, 0); // Without this, the Google logo is moved up by the value of edgePadding.bottom
  }
  ```

It worked! But I'm not sure if this is the right way to fix it.
  
However, the `fitToCoordinates` function does not respect the prop `mapPadding` passed to `<MapView />`, because the `baseMapPadding` is fixed at `50`, in this case we should consider what was passed by prop.

baseMapPadding fixed:
  https://github.com/react-native-community/react-native-maps/blob/master/lib/android/src/main/java/com/airbnb/android/react/maps/AirMapView.java#L85


In this line, the `baseMapPadding` should be the prop `mapPadding` passed to the `<MapView />`
https://github.com/react-native-community/react-native-maps/blob/master/lib/android/src/main/java/com/airbnb/android/react/maps/AirMapView.java#L888

Workaround to fix this problem:
```javascript
  this.map.fitToCoordinates(
    [bounds[0], bounds[1]],
    {
      edgePadding: {
        top: Platform.OS === "ios" ? 0 : PixelRatio.get() * paddingTop - 50, // 50 is the baseMapPaddinb https://github.com/react-native-community/react-native-maps/blob/master/lib/android/src/main/java/com/airbnb/android/react/maps/AirMapView.java#L85
        right: 0,
        left: 0,
        bottom:
          Platform.OS === "ios" ? 0 : PixelRatio.get() * paddintBottom - 50
      }
    },
    true
  );
```

**Comment by brmk** [comment]
@diegorodriguesvieira  your workaround helped me! thank you

**Comment by mausv** [comment]
I was having a similar issue using fitToCoordinates, found a workaround.

https://github.com/react-native-maps/react-native-maps/issues/2629#issuecomment-765001139

**Comment by hosle** [comment]
Regardless the mapPadding issue #3603 which is being fixed in another PR, it's due to the mapPadding is always set to (0, 0, 0, 0) at the end of `fitToCoordinates` method.
To solve this issue, we can reset the padding to the cache one in `AirMapView.java`

```
//AirMapView.java

  int baseLeftMapPadding;
  int baseRightMapPadding;
  int baseTopMapPadding;
  int baseBottomMapPadding;

// This is called by the mapPadding prop in AirMapViewManager.java
  public void applyBaseMapPadding(int left, int top, int right, int bottom){
    this.map.setPadding(left, top, right, bottom);
    baseLeftMapPadding = left;
    baseRightMapPadding = right;
    baseTopMapPadding = top;
    baseBottomMapPadding = bottom;
  }

 public void fitToCoordinates(ReadableArray coordinatesArray, ReadableMap edgePadding,
      boolean animated) {
...
    if (edgePadding != null) {
      map.setPadding(edgePadding.getInt("left") + baseLeftMapPadding,
              edgePadding.getInt("top") + baseTopMapPadding,
              edgePadding.getInt("right") + baseRightMapPadding,
              edgePadding.getInt("bottom") + baseBottomMapPadding);
    }
...
    // Move the google logo to the default base padding value.
    map.setPadding(baseLeftMapPadding, baseTopMapPadding, baseRightMapPadding, baseBottomMapPadding);
}

```

After fixed, it works as expected as iOS with the same usage in RN


<img src=https://user-images.githubusercontent.com/6284688/109633682-5449ff80-7b83-11eb-9f72-713b60fdcdc3.jpg width=50% height=50%/>

<img src=https://user-images.githubusercontent.com/6284688/109633696-57dd8680-7b83-11eb-9ba3-b5944de7226d.jpg width=50% height=50%/>

It's fixed in PR #3731 as well.

BTW, method `fitToSuppliedMarkers` may have the similar padding issue when setting padding without the original padding prop. But I haven't used this method, so didn't touch it in this case.
